src/Parsers/ASTTablesInSelectQuery.cpp

```diff
@@ -112,7 +113,7 @@ void ASTTableExpression::formatImpl(const FormatSettings & settings, FormatState
         settings.ostr << " ";
         database_and_table_name->formatImpl(settings, state, frame);
     }
-    else if (table_function)
+    else if (table_function && !table_function->as<ASTFunction>()->is_parameterized_view)
```

Comment:
as discussed, let's add `prefer_subquery_to_function_formatting`, we can also add assertion like `assert(!prefer_subquery_to_function_formatting || is_subquery)` or a fallback to printing `table_function` case in case `(!is_subquery && prefer_subquery_to_function_formatting)` if it is normal to not have a `subquery`.

Reply:
I have added this check in the if condition, so that we fallback to `table_function`, for parameterized_view this should never be the case, but seems like a good idea to fallback to `table_function` if subquery is not replaced for table function view, paramaterized view & maybe others in future.